src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -93,7 +89,11 @@ protected virtual IHandlerProvider GetHandlerProvider()
 
     protected virtual void SetupRequestDispatcher(IHandlerProvider handlerProvider)
     {
-        foreach (var metadata in handlerProvider.GetRegisteredMethods())
+        // Get unique set of methods from the handler provider independent of any language.
+        foreach (var metadata in handlerProvider
+            .GetRegisteredMethods()
+            .Select(m => new RequestHandlerMetadata(m.MethodName, m.RequestType, m.ResponseType))
```

Comment:
something seems odd here Mef-wise.  Can't put my finger on it.  But manually making metadata objects ssems off.

Reply:
This is not MEF metadata. It's also created in HandlerProvider.cs